### PR TITLE
Bubble sort

### PR DESCRIPTION
**Pull request by lekegitrepo**

### CODE REVIEW

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello @lekegitrepo @noomdalv 

:clap: Thank you that beautiful code :smile: 
There is one function that you forgot to implement `bubble_sort_by`.

Please kindly implement it and submit a new code review request.
Thanks

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      end
+    end
+    break unless swapped
+  end
+  p arr
+end
+
+bubble_sort_by(%w[hi hello hey]) do |left, right|
+  left.length - right.length
+end
```
Comment (samgaco) [inline bubble_sort.rb:31]: You forgot to take off this test case

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    swapped = false
+    n.times do |i|
+      if yield(arr[i], arr[i + 1]).positive?
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+        swapped = true
+      end
+    end
+    break unless swapped
+  end
+  p arr
```
Comment (samgaco) [inline bubble_sort.rb]: the "p" is not needed

**Review by samgaco — APPROVED** [review]
Hi @lekegitrepo and @noomdalv :high_brightness: 

The project is approved :white_check_mark: . You may merge to the master branch and submit the project completion form.

-----------------
**[Has to be addressed]** :warning: 
* May I ask about whether you are working in pairs or not? There is only one commit by @noomdalv , the projects should be done in pairs taking turns of 35 to 45 minutes. Please address this issue or contact student success if you are dealing with any issues.
-----------------
moving on bear this in mind :arrow_down: 

* **[For your next code review requests]**  Add a short summary of changes in Pull Request description

----------------------

* ###### STATUS OF THE PROJECT: APPROVED :heavy_check_mark:

**Pull request merged**